[msan] Unwind stack before fatal reports

vitalybuka opened the pull request:
Msan does not unwind stack in malloc without origins, but we still need trace
for fatal errors.

vitalybuka committed:
[msan] Unwind stack before fatal reports

Msan does not unwind stack in malloc without origins, but we still need trace
for fatal errors.

Pull Request: https://github.com/llvm/llvm-project/pull/77168

MaskRay commented on compiler-rt/lib/msan/msan.h:
```diff
@@ -321,6 +321,9 @@ const int STACK_TRACE_TAG_VPTR = STACK_TRACE_TAG_FIELDS + 1;
     stack.Unwind(pc, bp, nullptr, common_flags()->fast_unwind_on_fatal); \
   }
 
+#define GET_FATAL_STACK_TRACE \
```
Add a comment that this is to complement `GET_MALLOC_STACK_TRACE` when not tracking origins.

vitalybuka replied:
I think it's redundant here, we have similar macro in other sanitizers, and it's just shortcut for GET_FATAL_STACK_TRACE_PC_BP.

Maybe these way as in new snapshot?

MaskRay commented:
In some functions there is a parameter `*stack` and a variable `stack` in a scope. I wonder whether the variable name for `FATAL` should be changed to reduce confusion...

vitalybuka commented:
> In some functions there is a parameter `*stack` and a variable `stack` in `{}` that shadows the parameter. I wonder whether the variable name for `FATAL` should be changed to reduce confusion...

That's kind of convention for all sanitizers, if we have macro, it create `stack` local object.
I am not rely concerned with shadowing, at it's only for fatal cases, but if you strong about that, I'd rather change parameters in allocator.

vitalybuka committed:
[NFC][msan] Switch allocator interface to use BufferedStackTrace

We will need it to unwind for fatal errors.

Pull Request: https://github.com/llvm/llvm-project/pull/77363

vitalybuka committed:
rebase

Created using spr 1.3.4

MaskRay commented:
> I am not rely concerned with shadowing, at it's only for fatal cases, but if you strong about that, I'd rather change parameters in allocator.

Ah, I'm not concerned. You make the call whether the parameter or the local variable (both named `stack`) should be renamed to avoid confusion.

vitalybuka committed:
fix name

Created using spr 1.3.4
